github.go

```diff
@@ -201,48 +220,92 @@ func unauthenticatedGet(pages int, page int, client *http.Client) (GithubEvents,
 			Timeout: time.Second * 30,
 		}
 	}
-	r, err := httpClient.Get("https://api.github.com/events?page=" + strconv.Itoa(page))
+	r, err := httpClient.Get(url)
 	if err != nil {
 		log.Fatalf("Error in requesting data from API: %s", err.Error())
 	}
-	defer r.Body.Close()
 
-	RateLimitSpecs.Limit = parseHeader(r.Header, "x-ratelimit-limit")
-	RateLimitSpecs.Remaining = parseHeader(r.Header, "x-ratelimit-remaining")
-	RateLimitSpecs.ResetTimestamp = parseLongHeader(r.Header, "x-ratelimit-reset")
-	RateLimitSpecs.PollInterval = parseHeader(r.Header, "x-poll-interval") * pages
+	defer r.Body.Close()
 
 	if r.StatusCode == http.StatusNotModified { // (304) No new content
 		return nil, &APIError{"no new content", 304}
 	} else if r.StatusCode == http.StatusForbidden { // (403) Rate limit reached
 		return nil, &APIError{"rate limited", 403}
 	}
 
-	body, _ := ioutil.ReadAll(r.Body)
-	var events GithubEvents
-	json.Unmarshal(body, &events)
+	body, err := ioutil.ReadAll(r.Body)
 
-	return events, nil
+	if err != nil {
+		log.Fatal("Unable to read response body")
+		return nil, err
```

Comment:
As per the docs `Fatal is equivalent to Print() followed by a call to os.Exit(1)`, this `return` will never be called.